src/wp-includes/media.php

```diff
@@ -5614,3 +5834,45 @@ function wp_increase_content_media_count( $amount = 1 ) {
 
 	return $content_media_count;
 }
+
+/**
+ * Add fetchpriority='high' to $loading_attrs if applicable.
+ *
+ * @since 6.3.0
+ * @access private
+ *
+ * @param array[] $loading_attrs Array of the loading attributes for the element.
+ * @param array[] $attr Array of the attributes for the element.
+ * @return array[] $loading_attrs Updated loading attributes for the element.
+ */
+function wp_maybe_add_fetchpriority_high_attr( $loading_attrs, $attr ) {
+	if ( ! empty( $attr['fetchpriority'] ) ) {
+		$loading_attrs['fetchpriority'] = $attr['fetchpriority'];
+		return $loading_attrs;
+	}
+	$img_size                   = $attr['width'] * $attr['height'];
+	$wp_min_priority_img_pixels = apply_filters( 'wp_min_priority_img_pixels', 5000 );
+	if ( $wp_min_priority_img_pixels <= $img_size && wp_high_priority_element_flag() ) {
+		$loading_attrs['fetchpriority'] = 'high';
+		wp_high_priority_element_flag( false );
+	}
```

Comment:
Three small things to address here:
* The filter value has to be `50000` based on the requirements.
* More efficient to check the flag first before doing other calculations and checks.
* The filter needs documentation.
```suggestion
	// Bail if high-priority image was already encountered.
	if ( ! wp_high_priority_element_flag() ) {
		return $loading_attrs;
	}

	/**
	 * Filters the minimum square-pixels threshold for an image to be eligible as the high-priority image.
	 *
	 * @since 6.3.0
	 *
	 * @param int $threshold Minimum square-pixels threshold. Default 50000.
	 */
	$wp_min_priority_img_pixels = apply_filters( 'wp_min_priority_img_pixels', 50000 );
	if ( $wp_min_priority_img_pixels <= $attr['width'] * $attr['height'] ) {
		$loading_attrs['fetchpriority'] = 'high';
		wp_high_priority_element_flag( false );
	}
```